### PR TITLE
Export NxTextInput stateHelpers from index file - RSC-67

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-67

After this PR, importing the stateHelper functions directly from `stateHelpers.ts` will be considered deprecated

### CODE REVIEW

**Comment by rpokorny** [comment]
Adding more reviewers at @glenhunter 's request.

File: gallery/src/components/NxModal/NxModalFormErrorExample.tsx [hunk 1]
```diff
@@ -6,9 +6,11 @@
  */
 import React, {useState} from 'react';
 
-import {NxModal, NxFontAwesomeIcon, NxTextInput, NxLoadError, NxButton} from '@sonatype/react-shared-components';
+import {NxModal, NxFontAwesomeIcon, NxTextInput, NxLoadError, NxButton, nxTextInputStateHelpers}
+  from '@sonatype/react-shared-components';
 import {faAngry} from '@fortawesome/free-solid-svg-icons';
-import { initialState, userInput } from '@sonatype/react-shared-components/components/NxTextInput/stateHelpers';
+
+const { initialState, userInput } = nxTextInputStateHelpers;
```
Comment (raveneyex) [inline gallery/src/components/NxModal/NxModalFormErrorExample.tsx:13]:
```suggestion
const {initialState, userInput} = nxTextInputStateHelpers;
```

Comment (rpokorny) [reply gallery/src/components/NxModal/NxModalFormErrorExample.tsx:13]: > const {initialState, userInput} = nxTextInputStateHelpers;

That's not how I do it.

**Review by raveneyex — APPROVED** [review]

File: gallery/src/components/NxModal/NxModalFormErrorExample.tsx (hunk 1 above)
Comment (raveneyex) [reply gallery/src/components/NxModal/NxModalFormErrorExample.tsx:13]: Then line 9 and line 11 need be changed. For consistency.

Comment (rpokorny) [reply gallery/src/components/NxModal/NxModalFormErrorExample.tsx:13]: > Then line 9 and line 11 need be changed. For consistency.

Line 11 (and 7) are arguably out of scope but here you go: fd305b7b

Comment (raveneyex) [reply gallery/src/components/NxModal/NxModalFormErrorExample.tsx:13]: > > Then line 9 and line 11 need be changed. For consistency.
> 
> Line 11 (and 7) are arguably out of scope but here you go: [fd305b7](https://github.com/sonatype/sonatype-react-shared-components/commit/fd305b7b23d30bc3055d8361b8cf981f1cd2cbda)

Thanks for humoring my obsesssion!

**Review by marushkevych — APPROVED** [review]

**Pull request merged**